pkg/operator/ceph/controller.go

```diff
@@ -134,6 +135,8 @@ func (r *ReconcileConfig) reconcile(request reconcile.Request) (reconcile.Result
 		return opcontroller.ImmediateRetryResult, err
 	}
 
+	reconcileAllowLoopDevices(r.config.Parameters)
```

Comment:
Why is this needed? I could see that we would need to set the env var on the discovery daemon deployment spec, but not in the operator like this.

Reply:
This is related to https://github.com/rook/rook/pull/11268#discussion_r1014994697 .

For example, when the prepare job is created, we need to add an environment variable `CEPH_VOLUME_ALLOW_LOOP_DEVICES` to its pod definition **only when** loop devices are allowed by the new option `ROOK_CEPH_ALLOW_LOOP_DEVICES` in `rook-ceph-operator-config` ConfigMap. So `provisionOSDContainer` function needs to know the current value of `ROOK_CEPH_ALLOW_LOOP_DEVICES`.

In the current code, the `reconcile` function is the only place where the contents of the `rook-ceph-operator-config` ConfigMap is read. So I check the value of `ROOK_CEPH_ALLOW_LOOP_DEVICES` here and then use it when creating prepare pod. The new environment variable `ROOK_CEPH_ALLOW_LOOP_DEVICES` (which happens to have the same name as the new config) is just an intermediate variable.

Honestly, I feel it is clumsy. I will appreciate it if you have any ideas to improve it.

Reply:
Thank you for your comment.

I followed the pattern of `SetCephCommandsTimeout()` method and changed the code to use package variable again.

> But I'd rather not have a "reconcile" helper method if we're just setting a variable, so maybe call this directly:

I deleted `reconcileAllowLoopDevices` method.